### PR TITLE
Introduce style_guide to Keras governance.

### PR DESCRIPTION
**Pull request by LukeWood**
We can link to this in the future during PRs.

This PR is a working draft, I will continue to update it until we have a sufficiently in depth style guide.

### CODE REVIEW

File: style_guide.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+# Keras Style Guide
+
+This style guide contains information needed to write code in the style of the Keras
+codebase.
+
+### Layer-ish Functions
```
Comment (fchollet) [inline style_guide.md]: "Closures for Functional API building blocks"

File: style_guide.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+When writing Keras code it is common to create functions that apply a logical block of
+deep learning code, such as a ResNet block.  We write our blocks using a closure pattern
+that makes the behavior of these blocks mirror Keras layers.
+
+Bad:
```
Comment (fchollet) [inline style_guide.md]: "bad" seems too strong, the below code is fine. We can just show the closure and say "recommended"

File: style_guide.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+def DenseBlock(blocks, name=None):
+    """A dense block.
+    Args:
+      blocks: integer, the number of building blocks.
+      name: string, block label.
+    Returns:
+      a function that takes an input Tensor representing a DenseBlock.
+    """
+    if name is None:
+        name = f"dense_block_{backend.get_uid('dense_block')}"
```
Comment (fchollet) [inline style_guide.md:46]: Handling of name generation deserves its own section in the guide!

File: style_guide.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+tf.Variable(...)
+tf.reshape(...)
+keras.Input(...)
+keras.Model(...)
+keras.optimizers.Adam(...)
+layers.Layer(...)
+layers.Conv2D(...)
+```
+
+Note: do **not** use `import keras`. Use `from tensorflow import keras` instead.
```
Comment (fchollet) [inline style_guide.md:121]: Yes. (but note that we'll almost certainly reverse the guideline in the medium term future.)

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: style_guide.md (hunk 4 above)
Comment (LukeWood) [reply style_guide.md:121]: sounds good, for now we can prefer this and reverse it when the time comes.

Feel free to remove in your own pass when you do it.

**Pull request closed without merging**